pkg/genericapiserver/openapi/exceptions.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Methods in this file fix problems with generate OpenAPI spec. The ideal is to fix these problems in source and
+// remove these methods in future but for now we need these to generate a valid spec.
+
+package openapi
+
+import (
+	"bytes"
+	"github.com/go-openapi/spec"
+	"strings"
+	"unicode"
+)
+
+type PathOperationID struct {
+	path string
+	id   string
+}
+
+var opFix = map[PathOperationID]string{
+	PathOperationID{"/api/v1/configmaps", "listNamespacedConfigMap"}:   "listAllNamespacesConfigMap",
+	PathOperationID{"/api/v1/endpoints", "listNamespacedEndpoints"}:    "listAllNamespacesEndpoints",
+	PathOperationID{"/api/v1/events", "listNamespacedEvent"}:           "listAllNamespacesEvent",
+	PathOperationID{"/api/v1/limitranges", "listNamespacedLimitRange"}: "listAllNamespacesLimitRange",
+}
+
+// fixOperation tries to rename operation IDs to make OpenAPI spec valid. The best solution to repeated
+// ops is to fix it in the source. If that is not possible, the opFix map would let customize the renaming. If the ID
+// is not in the opFix map, the name will be concatenated with a string computed from the path. see inferNamePartFromPath
+// for more information.
```

Comment:
It should be done in api_installer. We should pass unique Operation name for each operation. There were just too many of them. I think it is better to have swagger 2.0 working and mark it as alpha, then fix those.